v2.0/partitioning.md

```diff
@@ -0,0 +1,455 @@
+
+{{site.data.alerts.callout_info}}Table partitioning is an <a href="enterprise-licensing.html">enterprise-only</a> feature. {{site.data.alerts.end}}
+
+<div id="toc"></div>
+
+## Overview
+
+Table partitioning helps you reduce latency and cost:
+
+- **Reduce latency**: Geo-partitioning allows you to keep user data close to the user, which reduces the distance that the data needs to travel, thereby reducing latency. To geo-partition a table, define location-based partitions while creating a table, create location-specific zone configurations, and apply the zone configurations to the corresponding partitions. 
+- **Reduce costs**: Archival-partitioning allows you to store infrequently-accessed data on slower and cheaper storage. To archival-partition a table, define frequency-based partitions while creating a table, create frequency-specific zone configurations with appropriate storage devices constraints, and apply the zone configurations to the corresponding partitions.
+
+## Define Table Partitions
+
+1. [Set the enterprise license key.](enterprise-licensing.html#set-the-trial-or-enterprise-license-key) 
+2. [Start the nodes with the right constraints](configure-replication-zones.html) (localities, storage devices, and other constraints).
+3. [Create table with partitions (and subpartitions)](partitioning.html#create-a-table-with-partitions) and [define the primary key.](partitioning.html#define-the-primary-key)
+4. [Create appropriate location-specific or storage-specific zone configurations and apply zone configurations to corresponding partitions](partitioning.html#set-zone-configurations)
+
+### Define the Primary Key
+
+A table can be partitioned based on either its primary key or secondary index. 
+
+The primary key required for partitioning is different from the conventional primary key. To define the primary key for partitioning, prefix the unique identifier(s) in the primary key with all columns you want to partition and subpartition the table on, in the order in which you want to nest your subpartitions. 
+
+#### Example
+
+Consider the database of a global online learning portal that has a table for students of all the courses across the world. If you want to geo-partition the table based on the countries of the students, then the primary key needs to be defined as:
+
+{% include copy-clipboard.html %} 
+~~~ sql
+> CREATE TABLE students (
+    id SERIAL,
+    name STRING,
+    email STRING,
+    country STRING,
+    expected_graduation_date DATE,   
+    PRIMARY KEY (country, id));
+~~~
+
+#### Partitioning Key Considerations
+
+- Once you set the primary key, you can’t change it later. Provision for all future subpartitions by including those columns in the primary key. In the example of the online learning portal, if you think you might want to subpartition based on `expected_graduation_date` in the future, define the primary key as `(country, expected_graduation_date, id)`.
```

Comment:
I think from a product perspective we need to note that this is on the roadmap. Asking someone to plan (potentially years in advance) the ways in which they may want to change the table is off brand. 

cc @awoods187 @dianasaur323

Reply:
+1 (but it's coming in 2.1)

Reply:
Yup -- let's just say that!